### PR TITLE
SoundWire: introduce controller_id 

### PR DESCRIPTION
**Pull request by plbossart**
Alternate solution to @krzk suggested patches on alsa-devel.

### CODE REVIEW

**Comment by plbossart** [comment]
@bardliao can you take a look if you have spare cycles. I don't have a local device with me so can't really smoke/stress test this change.

**Comment by bardliao** [comment]
> @bardliao can you take a look if you have spare cycles. I don't have a local device with me so can't really smoke/stress test this change.

@plbossart I tested it on a TGL RVP and it works. rt711's device name is sdw:0:0:025d:0711:00 with this PR, and playback/capture works.

File: drivers/soundwire/amd_manager.c [hunk 1]
```diff
@@ -926,6 +926,7 @@ static int amd_sdw_manager_probe(struct platform_device *pdev)
 	amd_manager->bus.compute_params = &amd_sdw_compute_params;
 	amd_manager->bus.clk_stop_timeout = 200;
 	amd_manager->bus.link_id = amd_manager->instance;
+	amd_manager->bus.controller_id = amd_manager->instance;
```
Comment (vijendarmukunda) [inline drivers/soundwire/amd_manager.c]: Currently we are populating one controller and two links under same controller scope.
for that we expect amd_manager->bus.controller_id should be set to zero. We can't populate as two separate controllers due to same design opted in other OS which can't be modified due to some design constraints.
This has been communicated during first version of amd soundwire patches upstream review.

In future any platform requires new design , then we will pass the controller id as driver data.

Comment (vijendarmukunda) [reply drivers/soundwire/amd_manager.c]: Please modify the commit message as well.

**Review by vijendarmukunda — CHANGES_REQUESTED** [review]
please change  controller id assignment as  amd_manager->bus.controller_id = 0;

File: drivers/soundwire/amd_manager.c (hunk 1 above)
Comment (plbossart) [reply drivers/soundwire/amd_manager.c]: thanks for the quick feedback @vijendarmukunda, much appreciated.

**Review by vijendarmukunda — APPROVED** [review]

**Comment by vijendarmukunda** [comment]
verified changes and it's working fine on AMD platforms.

**Review by bardliao — APPROVED** [review]

**Pull request merged**